### DISCUSSION TITLE
Composer plugin: find releases between two arbitrary versions

### DISCUSSION THREAD

**Discussion by andriokha** [composer/composer#12075] [Q&A / Support] [answered]
I'd like to update the plugin at https://github.com/pyrech/composer-changelogs.

Currently after an update eg. of `danielstjules/stringy` from `1.9.0` to `3.1.0` it will show release notes for the version updated to (last line):
```
 - danielstjules/stringy updated from 1.9.0 to 3.1.0 major
   See changes: https://github.com/danielstjules/Stringy/compare/1.9.0...3.1.0
   Release notes: https://github.com/danielstjules/Stringy/releases/tag/3.1.0
```

I'd like it to show release notes for all intemediate versions, eg.
```
 - danielstjules/stringy updated from 1.9.0 to 3.1.0 major
   See changes: https://github.com/danielstjules/Stringy/compare/1.9.0...3.1.0
   Release notes:
   - https://github.com/danielstjules/Stringy/releases/tag/3.1.0
   - https://github.com/danielstjules/Stringy/releases/tag/3.0.1
   - https://github.com/danielstjules/Stringy/releases/tag/3.0.0
   - https://github.com/danielstjules/Stringy/releases/tag/2.4.0
[snip]
   - https://github.com/danielstjules/Stringy/releases/tag/1.10.0
```

I was wondering if it's possible to use Composer to find out what releases exist between two particular versions? I've taken a look at the packages available in `\Composer\Plugin\PrePoolCreateEvent::getPackages()` and it doesn't necessarily include all versions required.

Thanks!

**Comment by Seldaek** [comment, marked as answer]
Nope it won't have all versions available at that point as we try to prune versions to only those that might  match for performance reasons. You'd have to poll the repositories again to fetch all versions of a package and see which are between the two you are updating.

**Reply by andriokha** [reply]
Thanks @Seldaek, appreciate your time.
